cl/llvm/clplug.cc

```diff
@@ -32,14 +32,27 @@ extern "C" {
 
 #include "llvm/Config/llvm-config.h"
 
-#if defined(LLVM_VERSION_MAJOR) && (LLVM_VERSION_MAJOR == 3) && (LLVM_VERSION_MINOR >= 7)
+#if defined(LLVM_VERSION_MAJOR) && ((LLVM_VERSION_MAJOR > 3) || \
+                                    ((LLVM_VERSION_MAJOR == 3) && (LLVM_VERSION_MINOR >= 7)))
 #   define LLVM_HOST_3_7_OR_NEWER
 #endif
 
+#if defined(LLVM_VERSION_MAJOR) && ((LLVM_VERSION_MAJOR > 3) || \
+                                    ((LLVM_VERSION_MAJOR == 3) && (LLVM_VERSION_MINOR >= 9)))
+#   define LLVM_HOST_3_9_OR_NEWER
+#endif
+
+#if defined(LLVM_VERSION_MAJOR) && (LLVM_VERSION_MAJOR >= 4)
+#   define LLVM_HOST_4_OR_NEWER
+#endif
+
+
+
 #ifdef LLVM_HOST_3_7_OR_NEWER
 #   include "llvm/Transforms/Scalar.h" // createLowerSwitchPass
 #   include "llvm/IR/LegacyPassManager.h"
 #   include "llvm/Transforms/IPO/PassManagerBuilder.h"
+#   include "llvm/Transforms/Utils.h"
```

Comment:
This header file for some reason does not exist on any of my installations of llvm, tried with 4.0.1, 5.0.2, and 6.0.1:
```
cl/llvm/clplug.cc:55:13: fatal error: llvm/Transforms/Utils.h: No such file or directory
 #   include "llvm/Transforms/Utils.h"
             ^~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I forgot to say that it compiles successfully with llvm-5.0.2 and llvm-6.0.1 if I just comment out the mentioned include directive.

Reply:
Sorry, my bad. I did not check the version of LLVM correctly. Should be fixed now.